crates/p2p_protocol/protocol/transit_encoding.capnp

```diff
@@ -0,0 +1,153 @@
+# lib3h In-Transit Encoding Handshake Protocol
+# These are all separate structs, rather than having a top-level root/union
+# This works because there is a well-defined handshake expectation
+# Anything unexpected is an error which should cause a connection drop
+
+@0xdf3e1d96ae657243;
+
+struct Halt {
+  # Let's at least try to be nice to people connecting to us
+  # Give them some info why we won't accept their connection
+
+  reasonCode @0 :ReasonCode;
+  reasonText @1 :Text;
+
+  enum ReasonCode {
+    unspecified @0;
+    badMagic @1;
+    badEncoding @2;
+    badNetworkId @3;
+    kxDecodeFail @4;
+    unexpectedSigPubKey @5;
+    badSignature @6;
+    encDecodeFail @7;
+  }
+}
+
+struct MsgConH1 {
+  # When opening a connection, you should send this first message
+
+  magic @0 :UInt16 = 0x0000;
+  # Protocol Identifier, this should be 0xa86c
+  # Note, default above is 0x0000, because otherwise 0xa86c would not be sent
+
+  encoding @1 :Encoding;
+  # How should we handle this stream (the remote may choose not to accept)
+
+  networkId @2 :Data;
+  # The network identifier we are trying to join
+  # If the remote has a different id, it may drop the connection.
+
+  kxPubKey @3 :Data;
+  # Send the remote our key exchange public key
+
+  enum Encoding {
+    # Which encoding scheme should we use
+
+    unknown @0;
+    # Default, you may get this if someone's using a newer protocol
+
+    openJson @1;
+    # Un-encrypted Json encoded
+
+    openPacked @2;
+    # Un-encrypted packed Capnproto encoded
+
+    sodiumJson @3;
+    # libsodium-encrypted Json encoded
+
+    sodiumPacked @4;
+    # libsodium-encrypted packed Capnproto encoded
+  }
+}
+
+struct MsgLsnH2 {
```

Comment:
Is there a reason why these struct names need to be so short?  Usually I'm ok with that, but `LsnH2` isn't transparent in meaning to me?

Reply:
+1

Reply:
hehe : ) Yeah, I can expand these a bit. Basically, one set of messages is coming from the connecting node ("Con") and the other set is coming from the listening node ("Lsn"), they are all numbered handshake messages, so handshake step 1 ("H1"), handshake step 2 ("H2"), etc.

Reply:
Alright, flipped this around and expanded: `MsgStep1FromConnect`, `MsgStep2FromListen`, etc